src/test/java/co/rsk/bitcoinj/core/TransactionWitnessTest.java

```diff
@@ -1,37 +1,53 @@
 package co.rsk.bitcoinj.core;
 
+import co.rsk.bitcoinj.crypto.TransactionSignature;
+import co.rsk.bitcoinj.params.MainNetParams;
+import co.rsk.bitcoinj.script.RedeemScriptUtils;
 import co.rsk.bitcoinj.script.Script;
 import co.rsk.bitcoinj.script.ScriptBuilder;
 import co.rsk.bitcoinj.script.ScriptOpCodes;
+import org.junit.Assert;
 import org.junit.Test;
-import org.spongycastle.util.encoders.Hex;
 
+import java.math.BigInteger;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 import static org.junit.Assert.*;
 
 public class TransactionWitnessTest {
-    private static final Script redeemScript = new Script(
-        Hex.decode("5221027de2af71862e0c64bf0ec5a66e3abc3b01fc57877802e6a6a81f6ea1d35610072102d9c67fef9f8d0707cbcca195eb5f26c6a65da6ca2d6130645c434bb924063856210346f033b8652a17d319d3ecbbbf20fd2cd663a6548173b9419d8228eef095012e53ae")
-    ); // data from tx https://mempool.space/testnet/tx/1744459aeaf7369aadc9fc40de9ab2bf575b14e35029b35a7ee4bbd3de65af7f
-    private static final byte[] redeemScriptHash = Sha256Hash.hash(redeemScript.getProgram());
-
+    public static final int FIRST_INPUT_INDEX = 0;
+    private static final NetworkParameters MAINNET_PARAMS = MainNetParams.get();
+    private static final List<BtcECKey> FEDERATION_KEYS = RedeemScriptUtils.getDefaultRedeemScriptKeys();
+    private static final BtcECKey fedKey1 = FEDERATION_KEYS.get(0);
+    private static final BtcECKey fedKey2 = FEDERATION_KEYS.get(1);
+    private static final BtcECKey fedKey3 = FEDERATION_KEYS.get(2);
+    private static final List<BtcECKey> ERP_FEDERATION_KEYS = RedeemScriptUtils.getEmergencyRedeemScriptKeys();
+    private static final long CSV_VALUE = 52_560L;
+    private static final Script redeemScript = RedeemScriptUtils.createP2shErpRedeemScript(
+        FEDERATION_KEYS,
+        ERP_FEDERATION_KEYS,
+        CSV_VALUE
+    );
+    private static final byte[] redeemScriptSerialized = redeemScript.getProgram();
+    private static final Script p2shOutputScript = ScriptBuilder.createP2SHP2WSHOutputScript(redeemScript);
```

Comment:
Careful with the naming, this can lead to confusion. I'd call it a p2sh-p2wsh output script

Reply:
done